lib/rspec/matchers/built_in/base_matcher.rb

```diff
@@ -112,6 +115,14 @@ def matcher_name
           end
         end
 
+        # @private
+        def reinitialize
+          @pairings_maximizer = nil
+          @best_solution = nil
+          @extra_items = nil
+          @missing_items = nil
```

Comment:
Those are ones from `contain_exactly`? Does it make sense to override this method there, and keeping it empty here?

Reply:
Yeah, that makes a lot more sense.  I'll pull this particular implementation into `contain_exactly` and leave it empty in the parent class 👍